lua/cmp_nvim_ultisnips/snippets.lua

```diff
@@ -5,12 +5,51 @@ local ft_snippets_cache = {}
 function M.load()
   if ft_snippets_cache[vim.bo.filetype] == nil then
     vim.F.npcall(vim.call, 'UltiSnips#SnippetsInCurrentScope', 1)
-    ft_snippets_cache[vim.bo.filetype] = vim.g.current_ulti_dict_info
+    local cur_dict_info = vim.g.current_ulti_dict_info
+    ft_snippets_cache[vim.bo.filetype] = M.add_regex_snippets(cur_dict_info)
   end
-
   return ft_snippets_cache[vim.bo.filetype]
 end
 
+-- Parses and returns snippets that include the 'r' option (UltiSnips does not return them).
+function M.add_regex_snippets(cur_dict_info)
+    local new_dict_info = {}
+    local visited_files = {}
+
+    for _, snippet_info in pairs(cur_dict_info) do
+      local snippet_path = string.gsub(snippet_info.location, ':%d*$', '')
+      if visited_files[snippet_path] == nil then
+        local content = vim.fn.readfile(snippet_path)
+        for line_idx, line in ipairs(content) do
+          -- skip non-regex snippets (specified by the snippet option r)
+          -- since they have already been added by UltiSnips
+          local _, _, trigger_word = string.find(line, '^snippet%s+"(.-)".*%s%w*r')
+          if trigger_word then
+            -- search for the last pair of quotes ("") to get the description
+            local _, _, description = string.find(line, '".*%s"(.*)".*$')
+            description = description or ''
+            local location = string.format('%s:%d', snippet_path, line_idx + 1)
```

Comment:
I am not sure here whether this should be `line_idx + 1` or `line_idx` since both seem to work. I would think only `line_idx ` should work since lua indices are 1-based. Maybe you can help with this ^^